### ISSUE TITLE
Steps component: accessibility bugfixes/improvements

### ISSUE THREAD

**Issue by MarcusMorba** [primefaces/primereact#2494]
### There is no guarantee in receiving an immediate response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeReact PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case (Bug Reports)**
Not required.

**Current behavior**

When checking the component https://www.primefaces.org/primereact/showcase/#/steps for accessibility issues with axe, several issues are reported e.g. in storybook or axe devtools browser extension.

1. General
- Nested interactive controls should be avoided for screen reader support. 

2. Basic (all steps) / Interactive (current step)
- a-tag has role=presentation: span-tag should be used instead - invalid role for a-tag. Alternative: use of <button> instead of <a>.

**Expected behavior**

Component should not show any accessibility issues. If used as a "tablist",  please see code example for "tablist" here: https://www.w3.org/TR/wai-aria-practices/#tabpanel

**Minimal reproduction of the problem with instructions**
Check https://www.primefaces.org/primereact/showcase/#/steps with axe DevTools extension.

**Please tell us about your environment:**

Windows 10

* **React version:**
n/a

* **PrimeReact version:**
7.0.1 - https://www.primefaces.org/primereact/showcase/#/steps

* **Browser:** 
all
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
n/a

**Comment by melloware** [comment]
Relates to this general ticket: https://github.com/primefaces/primereact/issues/1887
